convert_kinbody_to_urdf.py

```diff
@@ -0,0 +1,194 @@
+
+    def add_sphere(self, parent, radius):
+        geo = self.add_sub_element(parent, 'geometry')
+        self.add_sub_element(geo, 'sphere', {'radius': radius})
+
+    def add_mesh_render(self, parent, filename):
+        geo = self.add_sub_element(parent, 'geometry')
+        self.add_sub_element(geo, 'mesh', {'filename': filename})
+
+    def add_mesh_data(self, parent, val):
+        ws = val.split(' ')
+        geo = self.add_sub_element(parent, 'geometry')
+        if len(ws) == 1:
+            self.add_sub_element(geo, 'mesh', {'filename': ws[0]})
+        else:
+            self.add_sub_element(geo, 'mesh',
+                                 {'filename': ws[0],
+                                  'scale': '%.2f %.2f %.2f' % (ws[1], ws[1], ws[1])})
+
+    def add_color(self, parent, features):
+        if 'diffusecolor' in features:
+            val = features['diffusecolor']
+        elif 'ambientcolor' in features:
+            val = features['ambientcolor']
+        else:
+            val = '0.2 0.2 0.8'
+
+        if 'transparency' in features:
+            val += ' ' + features['transparency']
+        else:
+            val += ' 1.0'
+
+        mat = self.add_sub_element(parent, 'material', {'name': 'c'})
+        self.add_sub_element(mat, 'color', {'rgba': val})
+
+    def get_urdf(self):
+        if self.urdf is not None:
+            return self.urdf
+
+        self.urdf = ET.ElementTree()
+        urdf_root = ET.Element('robot')
+        urdf_root.attrib = self.kinbody.getroot().attrib
+        urdf_root.text = ''
+        self.link = self.add_sub_element(urdf_root, 'link',
+                                         {'name': 'base_link'})
+        self.convert_child(self.kinbody.getroot())
+        self.urdf._setroot(urdf_root)
+        return self.urdf
+
+    def print_tree(self, parent, level=0):
+        print(level, parent.tag, parent.attrib, parent.text.strip())
+        for child in parent:
+            self.print_tree(child, level + 1)
+
+
+def indent(elem, level=0):
+    i = "\n" + level * "  "
+    if len(elem):
+        if not elem.text or not elem.text.strip():
+            elem.text = i + "  "
+        if not elem.tail or not elem.tail.strip():
+            elem.tail = i
+        for elem in elem:
+            indent(elem, level + 1)
+        if not elem.tail or not elem.tail.strip():
+            elem.tail = i
+    else:
+        if level and (not elem.tail or not elem.tail.strip()):
+            elem.tail = i
+
+
+def script_main():
+    if len(sys.argv) != 2:
+        print('usage: python %s <filename>' % sys.argv[0])
+        return
+    filename = sys.argv[1]
+    if not os.path.isfile(filename):
+        print('Could not find file: %s' % filename)
+        return
+
+    kconv = Kinbody_Converter()
+    if not kconv.parse_from_file(filename):
+        return
+
+    urdf = kconv.get_urdf()
+
+    indent(urdf.getroot())
+
+    newfilename = filename[:-12] + '.urdf'
```

Comment:
Yeah, we could improve this part to take into account more general names. I would trim the extension using `os.path.splitext()`, and then optionally remove `.kinbody` if it exists at the end of the filename.

But this line works for the kinbody files in this repo as it is. So we could save this improvement for future changes. :smile: